### ISSUE TITLE
NumPy objects on the left of binary operations with `LightCurve` operate elementwise

### ISSUE THREAD

**Issue by dioph** [lightkurve/lightkurve#925]
#### Problem description

Right-side `LightCurve` binary operations fail with any NumPy left operand. Instead of using e.g. `LightCurve.__rmul__`, NumPy's `__mul__` takes precedence and attempts to operate elementwise. Although it's easy to circumvent this in practice by avoiding it altogether (i.e. by always using `LightCurve` objects on the left), my suggestion would be to set `__array_priority__` to force the interpreter to use `LightCurve` methods since it's an easy fix.

#### Example

```python
import lightkurve
import numpy as np
lc = lightkurve.LightCurve([1, 2, 3], [1, 1, 1])
print(np.float64(3) * lc)
print(np.std([1, 2, 3, 4]) + lc)
```

#### Expected behavior

The expected result in each case is a single `LightCurve` returned from `__radd__` or `__rmul__`, but instead we get an array of `LightCurve` objects, each with a single element of the product/sum. In particular, this behavior is not commutative (`np.float64(3) * lc != lc * np.float64(3)`).

#### Environment

-  platform (e.g. Linux, OSX, Windows): Linux
-  lightkurve version (e.g. 1.0b6): 1.11.3
-  installation method (e.g. pip, conda, source): pip

**Comment by barentsen** [comment]
Thank you for opening this issue @dioph!

There have been a number of changes related to these operations in Lightkurve v2.0 (i.e. the current main branch under development). I am seeing the current behavior:

<img width="1004" alt="Screen Shot 2021-01-07 at 11 53 04 AM" src="https://user-images.githubusercontent.com/817669/103938365-f814b300-50de-11eb-85bc-7cae0c7ba6ea.png">

...so left-hand multiplication with a float appears to work, but we run into a funny TypeError when passing a numpy type.  That ought to be fixed for sure!

**Comment by warrickball** [comment]
For what it's worth, these examples work when the `LightCurve` is the first argument. e.g. with
```python
import lightkurve
lc = lightkurve.LightCurve(time=[1, 2, 3], flux=[1, 1, 1], flux_err=[1, 1, 1])
```
``np.float64(3) * lc`` gives me
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-23-a877593692dd> in <module>
----> 1 np.float64(3) * lc

TypeError: The DTypes <class 'numpy.dtype[void]'> and <class 'numpy.dtype[float16]'> do not have a common DType. For example they cannot be stored in a single array unless the dtype is `object`.
```
whereas `lc * np.float64(3)` prints the expected output:
```
<LightCurve length=3>
 time    flux  flux_err
                       
object float64 float64 
------ ------- --------
   1.0     3.0      3.0
   2.0     3.0      3.0
   3.0     3.0      3.0
```
This stretches my knowledge of Python but I think this is because the NumPy type tries to resolve the operation but raises a TypeError instead of a NotImplementedError, so the Lightkurve object doesn't get a chance.

**Comment by warrickball** [comment]
> ... my suggestion would be to set `__array_priority__` to force the interpreter to use `LightCurve` methods since it's an easy fix.

This works. If I add
```python
    __array_priority__ = -999_999 # anything > -1_000_000 seems to work
```
before `def LightCurve.__init__` then I get the same result no matter the order of operands. Happy to open another PR. :)

**Comment by barentsen** [comment]
I opened a PR to fix this issue (#925).

@warrickball Thanks so much for figuring out the role of `__array_priority__`.  That made fixing this issue really issue!  It would probably have taken me a long time to discover this.

**Comment by barentsen** [comment]
The fix for this issue will be part of the upcoming v2.0.10 release.
